moveit_planners/pilz_industrial_motion_planner/src/trajectory_functions.cpp

```diff
@@ -198,13 +198,17 @@ bool pilz_industrial_motion_planner::generateJointTrajectory(
   ros::Time generation_begin = ros::Time::now();
 
   // generate the time samples
-  const double epsilon = 10e-06;  // avoid adding the last time sample twice
-  std::vector<double> time_samples;
-  for (double t_sample = 0.0; t_sample < trajectory.Duration() - epsilon; t_sample += sampling_time)
+  std::vector<double> time_samples = { 0.0 };
+  double actual_sampling_time = sampling_time;
+  if (trajectory.Duration() > 10e-06)
```

Comment:
Why exclude these ultra-short trajectories at all?
The timing is always relative to the robot: For a fast-moving one, 10^-6 might be a long time to move within, a slow one will probably barely movel

Reply:
I was just trying to preserve the existing behaviour; previously if `trajectory.Duration()` was less than `epsilon`, the trajectory would only contain one point.

If you think the old behaviour needs updating, I am happy to do so.